File: src/main/java/io/galagutskiy/idea/plugin/symbol/SymbolTemplateContextType.java
package io.galagutskiy.idea.plugin.symbol;

import com.intellij.codeInsight.template.TemplateContextType;
import com.intellij.psi.PsiFile;
import org.jetbrains.annotations.NotNull;

public class SymbolTemplateContextType extends TemplateContextType {

    protected SymbolTemplateContextType() {
        super("SYMBOL", "Symbol");
    }

    @Override
    public boolean isInContext(@NotNull PsiFile file, int offset) {
        return file.isWritable();
    }
}

File: src/main/kotlin/io/teinnsei/idea/plugin/symbol/SymbolTemplateContextType.kt
package io.teinnsei.idea.plugin.symbol

import com.intellij.codeInsight.template.TemplateContextType
import com.intellij.psi.PsiFile
import org.jetbrains.annotations.Nls
import org.jetbrains.annotations.NotNull

abstract class SymbolTemplateContextType(id: @NotNull String, presentableName: @Nls @NotNull String)
    : TemplateContextType(id, presentableName) {

    final override fun isInContext(file: PsiFile, offset: Int): Boolean {
        return file.isWritable
    }
}